blackjack/blackjack.go

```diff
@@ -0,0 +1,236 @@
+package blackjack
+
+import (
+	"cards"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+type Action int
+
+const (
+	None Action = iota
+	Stand
+	Hit
+)
+
+type Outcome int
+
+const (
+	PlayerBlackjack Outcome = iota
+	PlayerWin
+	PlayerLose
+	PlayerTie
+	PlayerBust
+)
+
+type Game struct {
+	Player Player
+	Dealer Player
+	Shoe   cards.Deck
+}
+
+type Option func(*Game) error
+
+func WithCustomDeck(deck cards.Deck) Option {
+	return func(g *Game) error {
+		g.Shoe = deck
+		return nil
+	}
+}
+
+func NewBlackjackGame(opts ...Option) (*Game, error) {
+	deck := cards.NewDeck(
+		cards.WithNumberOfDecks(1),
+	)
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	game := &Game{
+		Shoe: deck.Shuffle(random),
+	}
+
+	for _, o := range opts {
+		o(game)
+	}
+
+	fmt.Println(game.Shoe.Cards)
+	return game, nil
+}
+
+func (g *Game) Deal() cards.Card {
+	var card cards.Card
+
+	card, g.Shoe.Cards = g.Shoe.Cards[0], g.Shoe.Cards[1:]
+	g.Shoe.Cards = append(g.Shoe.Cards, card)
+
+	return card
+
+}
+
+func (g *Game) RunCLI() {
+
+	for {
+		g.Player.Hand = []cards.Card{}
+		g.Dealer.Hand = []cards.Card{}
+		g.Player.Action = None
+		g.Start()
+		g.DealerStart()
+		outcome := g.Outcome()
+		fmt.Println(ReportWinner(outcome))
+	}
+}
+
+func (g *Game) Start() {
+
+	fmt.Println("")
+	fmt.Println("****** NEW GAME ******")
+
+	g.Player.GetCard(g.Deal())
+	g.Dealer.GetCard(g.Deal())
+	g.Player.GetCard(g.Deal())
+	g.Dealer.GetCard(g.Deal())
+
+	fmt.Println(g.Dealer.DealerString())
+	fmt.Println("Player has " + g.Player.String())
+
+	if g.Player.Score() == 21 {
+		g.Player.Action = Stand
+	}
+
+	for g.Player.Action != Stand {
+
+		g.SetPlayerAction()
+
+		if g.Player.Action == Hit {
+			g.Player.GetCard(g.Deal())
+			g.Player.Action = None
+
+			fmt.Println(g.Player.String())
+			if g.Player.Score() > 21 {
+				g.Player.Action = Stand
```

Comment:
It doesn't seem like this should be necessary: if the player has bust, there are no more decisions for them to make!